src/multiharp/device.rs

```diff
@@ -192,13 +194,58 @@ pub struct MH160Device<T: MhlibWrapper> {
     mhlib_wrapper: T,
 }
 
+mod defaults {
+    pub fn default_true() -> i32 {
+        1
+    }
+    pub fn default_false() -> i32 {
+        0
+    }
+}
+
+#[allow(clippy::unsafe_derive_deserialize)]
+#[cfg_attr(feature = "python", pyclass(get_all, set_all))]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct RowEventFilterConfig {
+    pub row_filters: Vec<RowFilter>,
+}
+
+#[allow(clippy::unsafe_derive_deserialize)]
+#[cfg_attr(feature = "python", pyclass(get_all, set_all))]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct RowFilter {
+    pub time_range_ps: i32,
+    #[serde(default = "defaults::default_false")]
+    pub invert: i32,
+    #[serde(default)]
+    pub pass_channels: Vec<MH160ChannelId>,
+    pub use_channels: Vec<MH160ChannelId>,
+    pub match_count: i32,
+}
+
+#[allow(clippy::unsafe_derive_deserialize)]
+#[cfg_attr(feature = "python", pyclass(get_all, set_all))]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct MainEventFilterConfig {
+    #[serde(default = "defaults::default_true")]
+    pub enable: i32,
+    pub time_range_ps: i32,
+    #[serde(default = "defaults::default_false")]
+    pub invert: i32,
```

Comment:
This doesn't feel like the best way to solve the problem of translating boolean values into `i32`s.

This is partially my fault for suggesting the use of enums instead of boolean values. That might not have been the right idea.